### PR TITLE
LOK-2459 : Syslog in Classic to Minion.

### PR DESCRIPTION
**Pull request by MusaidAli**
## Description
Syslog in Classic to Minion .Review Syslog in classic and shift to Lokahi .

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2459

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: minion/minion-syslog/listener/pom.xml [hunk 1]
```diff
@@ -67,5 +67,27 @@
       <artifactId>camel-core-languages</artifactId>
       <version>${camel.version}</version>
     </dependency>
+      <dependency>
+          <groupId>org.opennms.lokahi.shared</groupId>
+          <artifactId>ipc-api</artifactId>
+          <version>0.1.0-SNAPSHOT</version>
```
Comment (cgorantla) [inline minion/minion-syslog/listener/pom.xml:73]: try to use project version

File: shared-lib/protobuf/src/main/proto/syslog-config.proto [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ *******************************************************************************/
+syntax = "proto3";
+
+import "google/protobuf/any.proto";
+
+package opennms.sink.syslog;
+option java_multiple_files = true;
+option java_package = "org.opennms.sink.syslog.contract";
+
+message SyslogConfig {
```
Comment (cgorantla) [inline shared-lib/protobuf/src/main/proto/syslog-config.proto:36]: Don't assume that whatever works for TrapsConfig works for Syslog, Please try to understand syslog config from Classic

File: minion/minion-syslog/listener/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 3]
```diff
@@ -1,28 +1,60 @@
...
+           http://camel.apache.org/schema/blueprint http://camel.apache.org/schema/blueprint/camel-blueprint.xsd
+		   http://aries.apache.org/blueprint/xmlns/blueprint-cm/v1.3.0
+		   http://aries.apache.org/schemas/blueprint-cm/blueprint-cm-1.3.0.xsd">
+
+
+    <reference id="messageDispatcherFactory" interface="org.opennms.horizon.shared.ipc.sink.api.MessageDispatcherFactory"/>
+    <reference id="identity" interface="org.opennms.horizon.shared.ipc.rpc.IpcIdentity"/>
+
+
+    <cm:property-placeholder id="syslogProperties"  persistent-id="org.opennms.horizon.minion.syslog.listener.syslog" update-strategy="reload">
```
Comment (cgorantla) [inline minion/minion-syslog/listener/src/main/resources/OSGI-INF/blueprint/blueprint.xml:16]: we need to derive config from Inventory,  check how TrapConfig ( static for now) is send from Inventory service. see `TrapConfigService.java`

**Review by cgorantla — CHANGES_REQUESTED** [review]

File: shared-lib/protobuf/src/main/proto/syslog-config.proto (hunk 2 above)
Comment (MusaidAli) [reply shared-lib/protobuf/src/main/proto/syslog-config.proto:36]: removed snmpUser ,other configs added.

File: minion/minion-syslog/listener/src/main/resources/OSGI-INF/blueprint/blueprint.xml (hunk 3 above)
Comment (MusaidAli) [reply minion/minion-syslog/listener/src/main/resources/OSGI-INF/blueprint/blueprint.xml:16]: added task for syslog ,this should be seen in console like trap and flow config.currently not showing.any idea

Comment (MusaidAli) [reply minion/minion-syslog/listener/src/main/resources/OSGI-INF/blueprint/blueprint.xml:16]: It shows in inventory logs but not in minion console logs .

Comment (MusaidAli) [reply minion/minion-syslog/listener/src/main/resources/OSGI-INF/blueprint/blueprint.xml:16]: resolved.

**Comment by fooker** [comment]
Has this PR been split up? It looks like it contains a lot of stuff similar to other PRs from you.

**Comment by smunir-onms** [comment]
Covered under https://github.com/OpenNMS-Cloud/lokahi/pull/1887

**Pull request closed without merging**